### PR TITLE
feat: route crd add timeout fields

### PR DESCRIPTION
**Pull request by Donghui0**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

### New feature or improvement
- Describe the details and related test reports.
Route.Spec add plugin_config_id And timeout fields. 
___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

File: pkg/kube/apisix/apis/config/v2alpha1/types.go [hunk 1]
```diff
@@ -99,6 +100,7 @@ type ApisixRouteHTTP struct {
 	Backends       []*ApisixRouteHTTPBackend  `json:"backends" yaml:"backends"`
 	Websocket      bool                       `json:"websocket" yaml:"websocket"`
 	Plugins        []*ApisixRouteHTTPPlugin   `json:"plugins,omitempty" yaml:"plugins,omitempty"`
+	PluginConfigID string                     `json:"plugin_config_id,omitempty" yaml:"plugin_config_id,omitempty"`
```
Comment (tokers) [inline pkg/kube/apisix/apis/config/v2alpha1/types.go:103]: IMHO this is a terrible way to expose the ID of an APISIX configuration in the declarative configuration way, it's not easy for people to configure it.

Another CRD to group a bunch of plugins can be introduced, say `ApisixPluginConfig`, and here we should specify the name of such CR.

**Review by tao12345666333 — CHANGES_REQUESTED** [review]
you can split into two PR

* add Timeout
* add PluginConfig CR

**Comment by tao12345666333** [comment]
@Donghui0 

https://github.com/apache/apisix-ingress-controller/issues/638

**Comment by tao12345666333** [comment]
@Donghui0  please check the unit-test errors. https://github.com/apache/apisix-ingress-controller/pull/609/checks?check_run_id=3358610706

**Comment by tao12345666333** [comment]
@Donghui0 yep, I re-run the job，no more errors.

**Review by tao12345666333 — APPROVED** [review]

**Comment by tao12345666333** [comment]
ping @tokers @gxthrj for review

**Review by tokers — APPROVED** [review]

**Pull request merged**